src/mito2/src/read/unordered_scan.rs

```diff
@@ -228,6 +228,11 @@ impl RegionScanner for UnorderedScan {
 
         Ok(stream)
     }
+
+    fn has_predicate(&self) -> bool {
+        let predicate = self.stream_ctx.input.predicate();
```

Comment:
Does it contain the time predicate?

Reply:
It's derived from filters in scan request directly.

Reply:
We removes the time filter from the request so it may be empty if there is only a filter for the timestamp